From 6f022cc47236c4aec17151cb3bc4a7e25cc5ebaf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dirk=20M=C3=BCller?= <dmueller@suse.com>
Date: Fri, 10 Nov 2023 10:06:35 +0100
Subject: [PATCH] [PATCH] debrand kernel
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Dirk Müller <dmueller@suse.com>
---
 SPECS/kernel.spec | 32 +++++++++++++++-----------------
 1 file changed, 15 insertions(+), 17 deletions(-)

diff --git a/SPECS/kernel.spec b/SPECS/kernel.spec
index b6f9c53..1ba7081 100644
--- a/SPECS/kernel.spec
+++ b/SPECS/kernel.spec
@@ -810,18 +810,14 @@ Source1: Makefile.rhelver
 %define secureboot_ca_0 %{_datadir}/pki/sb-certs/secureboot-ca-%{_arch}.cer
 %define secureboot_key_0 %{_datadir}/pki/sb-certs/secureboot-kernel-%{_arch}.cer
 
-%if 0%{?centos}
-%define pesign_name_0 centossecureboot201
-%else
 %ifarch x86_64 aarch64
-%define pesign_name_0 redhatsecureboot501
+%define pesign_name_0 openelabootsigningcert
 %endif
 %ifarch s390x
-%define pesign_name_0 redhatsecureboot302
+%define pesign_name_0 openelabootsigningcert
 %endif
 %ifarch ppc64le
-%define pesign_name_0 redhatsecureboot701
-%endif
+%define pesign_name_0 openelabootsigningcert
 %endif
 
 # signkernel
@@ -901,8 +897,8 @@ Source84: mod-internal.list
 Source85: mod-partner.list
 Source86: mod-kvm.list
 
-Source100: rheldup3.x509
-Source101: rhelkpatch1.x509
+Source100: openeladup1.x509
+Source101: openelakpatch1.x509
 Source102: rhelimaca1.x509
 Source103: rhelima.x509
 Source104: rhelima_centos.x509
@@ -962,6 +958,7 @@ Patch1: patch-%{patchversion}-redhat.patch
 
 # empty final patch to facilitate testing of kernel patches
 Patch999999: linux-kernel-test.patch
+Patch1000000: 1000-Debrand-specific-Red-Hat-messages.patch
 
 # END OF PATCH DEFINITIONS
 
@@ -1245,10 +1242,10 @@ Summary: gcov graph and source files for coverage data collection.\
 %{nil}
 
 %package -n kernel-abi-stablelists
-Summary: The Red Hat Enterprise Linux kernel ABI symbol stablelists
+Summary: The OpenELA kernel ABI symbol stablelists
 AutoReqProv: no
 %description -n kernel-abi-stablelists
-The kABI package contains information pertaining to the Red Hat Enterprise
+The kABI package contains information pertaining to the OpenELA
 Linux kernel ABI, including lists of kernel symbols that are needed by
 external Linux kernel modules, and a yum plugin to aid enforcement.
 
@@ -1258,7 +1255,7 @@ Summary: The baseline dataset for kABI verification using DWARF data
 Group: System Environment/Kernel
 AutoReqProv: no
 %description kernel-kabidw-base-internal
-The package contains data describing the current ABI of the Red Hat Enterprise
+The package contains data describing the current ABI of the OpenELA
 Linux kernel, suitable for the kabi-dw tool.
 %endif
 
@@ -1358,7 +1355,7 @@ Requires: kernel%{?1:-%{1}}-modules-core-uname-r = %{KVERREL}%{uname_suffix %{?1
 AutoReq: no\
 AutoProv: yes\
 %description %{?1:%{1}-}modules-internal\
-This package provides kernel modules for the %{?2:%{2} }kernel package for Red Hat internal usage.\
+This package provides kernel modules for the %{?2:%{2} }kernel package for OpenELA internal usage.\
 %{nil}
 
 %if %{with_realtime}
@@ -1697,6 +1694,7 @@ cp -a %{SOURCE1} .
 ApplyOptionalPatch patch-%{patchversion}-redhat.patch
 %endif
 
+ApplyOptionalPatch 1000-Debrand-specific-Red-Hat-messages.patch
 ApplyOptionalPatch linux-kernel-test.patch
 
 # END OF PATCH APPLICATIONS
@@ -1790,7 +1788,7 @@ done
 # Adjust FIPS module name for RHEL
 %if 0%{?rhel}
 for i in *.config; do
-  sed -i 's/CONFIG_CRYPTO_FIPS_NAME=.*/CONFIG_CRYPTO_FIPS_NAME="Red Hat Enterprise Linux %{rhel} - Kernel Cryptographic API"/' $i
+  sed -i 's/CONFIG_CRYPTO_FIPS_NAME=.*/CONFIG_CRYPTO_FIPS_NAME="OpenELA %{rhel} - Kernel Cryptographic API"/' $i
 done
 %endif
 
@@ -2428,12 +2426,12 @@ BuildKernel() {
 %if %{efiuki}
     if [ "$Variant" != "rt" ] && [ "$Variant" != "rt-debug" ]; then
         popd
-    
+
         KernelUnifiedImageDir="$RPM_BUILD_ROOT/lib/modules/$KernelVer"
         KernelUnifiedImage="$KernelUnifiedImageDir/$InstallName-virt.efi"
-    
+
         mkdir -p $KernelUnifiedImageDir
-    
+
         dracut --conf=%{SOURCE150} \
                --confdir=$(mktemp -d) \
                --verbose \
-- 
2.42.0

